File: src/test/java/io/binac/leetcode/BinaryTreeInorderTraversalTest.java
package io.binac.leetcode;

import io.binac.leetcode.util.TreeNode;
import io.binac.leetcode.util.Trees;
import org.junit.jupiter.api.Test;

import java.util.Arrays;
import java.util.List;

import static org.junit.jupiter.api.Assertions.*;

class BinaryTreeInorderTraversalTest {
    private final BinaryTreeInorderTraversal.Solution1 solution1 = new BinaryTreeInorderTraversal.Solution1();

    @Test
    void test1() {
        TreeNode root = Trees.asTree(1, null, 2, 3);
        List<Integer> expected = Arrays.asList(1, 3, 2);
        assertEquals(expected, solution1.inorderTraversal(root));
    }
}

File: cpp/unittests/BinaryTreeInorderTraversalTest.cpp
#include "BinaryTreeInorderTraversal.h"
#include "gtest/gtest.h"

using namespace lcpp;

TEST(BinaryTreeInorderTraversal, Solution94_1) {
  const auto Root = newTree({1, NTNode, 2, 3});
  const std::vector<int> Expected{1, 3, 2};
  EXPECT_EQ(Expected, Solution94_1().inorderTraversal(Root));
  deleteTree(Root);
}